scopes/dependencies/dependencies/dependencies-loader/auto-detect-deps.ts

```diff
@@ -578,34 +590,28 @@ export class AutoDetectDeps {
     this.debugDependenciesData.unidentifiedPackages = unidentifiedPackages;
   }
 
-  private _pushToDependenciesIfNotExist(
-    dependency: Dependency,
-    fileType: FileType,
-    depDebug: DebugComponentsDependency
-  ) {
+  private _pushToDependenciesIfNotExist(dependency: Dependency, opts: PushToDepsArrayOpts) {
     const existingDependency = this.getExistingDependency(this.allDependencies.dependencies, dependency.id);
     const existingDevDependency = this.getExistingDependency(this.allDependencies.devDependencies, dependency.id);
     // no need to enter dev dependency to devDependencies if it exists already in dependencies
-    if (existingDependency || (existingDevDependency && fileType.isTestFile)) {
+    if (existingDependency || (existingDevDependency && opts.fileType.isTestFile)) {
       return;
     }
     // at this point, either, it doesn't exist at all and should be entered.
     // or it exists in devDependencies but now it comes from non-dev file, which should be entered
     // as non-dev.
-    this.pushToDependenciesArray(dependency, fileType, depDebug);
+    this.pushToDependenciesArray(dependency, opts);
   }
 
-  private pushToDependenciesArray(
-    currentComponentsDeps: Dependency,
-    fileType: FileType,
-    depDebug: DebugComponentsDependency
-  ) {
-    if (fileType.isTestFile) {
+  private pushToDependenciesArray(currentComponentsDeps: Dependency, opts: PushToDepsArrayOpts) {
+    if (opts.fileType.isTestFile) {
       this.allDependencies.devDependencies.push(currentComponentsDeps);
+    } else if (opts.isPeer) {
+      this.allDependencies.peerDependencies.push(currentComponentsDeps);
```

Comment:
if it's used from a test file, should we make it dev dep or peer dep here?
that's a question. not sure about it.

Reply:
I think it should be a dev dep in that case. Test files are not used when the package is installed as a dependency, which is when peer deps are useful.